jvm/core/src/main/scala/org/scalatest/tools/SuiteSortingReporter.scala

```diff
@@ -173,6 +173,7 @@ private[scalatest] class SuiteSortingReporter(dispatch: Reporter, val testSortin
           dispatch(doneEvent)
           dispatchToRegisteredSuiteReporter(head.suiteId, doneEvent)
         }
+        suiteReporterMap -= (head.suiteId)
```

Comment:
@johnduffell I am not sure if this is needed, it seems like you have also added the same code in the fireReadySuiteEvents which is called right after this line, perhaps I am missing something here?

Reply:
same comment as https://github.com/scalatest/scalatest/pull/1843#pullrequestreview-451555737
> `fireReadySuiteEvents` is only called with the tail of the `slotListBuf`, we are only dealing with the head by the looks of it in this function.
> I may well have not fully understood the code, but it looked like I needed to cover both.